quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/JavaTextDocumentService.java

```diff
@@ -0,0 +1,138 @@
+import com.redhat.quarkus.settings.SharedSettings;
+
+/**
+ * LSP text document service for Java file.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class JavaTextDocumentService extends AbstractTextDocumentService {
+
+	private final QuarkusLanguageServer quarkusLanguageServer;
+	private final SharedSettings sharedSettings;
+
+	public JavaTextDocumentService(QuarkusLanguageServer quarkusLanguageServer, SharedSettings sharedSettings) {
+		this.quarkusLanguageServer = quarkusLanguageServer;
+		this.sharedSettings = sharedSettings;
+	}
+
+	@Override
+	public void didOpen(DidOpenTextDocumentParams params) {
+
+	}
+
+	@Override
+	public void didChange(DidChangeTextDocumentParams params) {
+
+	}
+
+	@Override
+	public void didClose(DidCloseTextDocumentParams params) {
+
+	}
+
+	@Override
+	public void didSave(DidSaveTextDocumentParams params) {
+
+	}
+
+	@Override
+	public CompletableFuture<List<? extends CodeLens>> codeLens(CodeLensParams params) {
+		boolean urlCodeLensEnabled = sharedSettings.getCodeLensSettings().isUrlCodeLensEnabled();
+
+		if (!urlCodeLensEnabled) {
+			// Don't consume JDT LS extension if all code lens are disabled.
+			return CompletableFuture.completedFuture(Collections.emptyList());
+		}
+		QuarkusJavaCodeLensParams javaParams = new QuarkusJavaCodeLensParams(params.getTextDocument().getUri());
+		if (sharedSettings.getCommandCapabilities().isCommandSupported(CommandKind.COMMAND_OPEN_URI)) {
+			javaParams.setOpenURICommand(CommandKind.COMMAND_OPEN_URI);
+		}
+		javaParams.setCheckServerAvailable(true);
+		javaParams.setUrlCodeLensEnabled(urlCodeLensEnabled);
+		// javaParams.setLocalServerPort(8080); // TODO : manage this server port from
+		// the settings
+		return quarkusLanguageServer.getLanguageClient().quarkusJavaCodelens(javaParams);
+	}
+
+	public void updateCodeLensSettings(QuarkusCodeLensSettings newCodeLens) {
+		sharedSettings.getCodeLensSettings().setUrlCodeLensEnabled(newCodeLens.isUrlCodeLensEnabled());
+	}
+
+
+	@Override
+	public CompletableFuture<Hover> hover(TextDocumentPositionParams params) {
+		// TextDocument textDocument = params.getTextDocument();
+		boolean markdownSupported =  sharedSettings.getHoverSettings().isContentFormatSupported(MarkupKind.MARKDOWN);
+		QuarkusJavaHoverParams javaParams = new QuarkusJavaHoverParams(params.getTextDocument().getUri(), params.getPosition());
+		return quarkusLanguageServer.getLanguageClient().quarkusJavaHover(javaParams).thenApply(info -> {
+
+			if (info == null || !Ranges.containsPosition(info.getRange(), params.getPosition())) {
+				return null;
+			}
+
+			String hoverContent = createHoverContent(info.getPropertyKey(), info.getPropertyValue(), markdownSupported);
+			MarkupContent content = new MarkupContent("markdown", hoverContent);
+			Hover hover = new Hover(content, info.getRange());
+			return hover;
+		});
+	}
+
+	private String createHoverContent(String propertyKey, String propertyValue, boolean markdownSupported) {

```

Comment:
Move this method in DocumentationUtils#getDocumentation(String propertyKey, String propertyValue, boolean markdownSupported) which must return MarkupContent (see existing DocumentationUtils#getDocumentation(EnumItem item, boolean markdown))